pkg/controller/telemetry/map_metric.go

```diff
@@ -0,0 +1,275 @@
+/*
+* Copyright The Kmesh Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at:
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"time"
+
+	"github.com/cilium/ebpf"
+)
+
+const (
+	mapMetricFlushInterval = 5 * time.Second
+)
+
+type MapMetricController struct {
+	mapMetricCache map[mapMetricLabels]*mapUsageInfo
+}
+
+type mapUsageMetric struct {
+	mapId      uint32
+	mapName    string
+	mapType    string
+	maxEntries uint32
+	keySize    uint32
+	valueSize  uint32
+	memLock    uint32
+	entryCount uint32
+}
+
+type mapUsageInfo struct {
+	memLock    uint32
+	entryCount uint32
+	maxEntries uint32
+}
+
+type mapMetricLabels struct {
+	mapId        string
+	mapName      string
+	mapType      string
+	podName      string
+	podNamespace string
+}
+
+func NewMapMetric() *MapMetricController {
+	return &MapMetricController{
+		mapMetricCache: map[mapMetricLabels]*mapUsageInfo{},
+	}
+}
+
+func (m *MapMetricController) Run(ctx context.Context) {
+	log.Infof("map Run 1111111")
+	if m == nil {
+		return
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				// Metrics updated every 5 seconds
+				time.Sleep(mapMetricFlushInterval)
+				m.updatePrometheusMetric()
+			}
+		}
+	}()
+}
+
+func (m *MapMetricController) buildMapMetric(data *mapUsageMetric) mapMetricLabels {
+	labels := mapMetricLabels{}
+	// Get the actual pod name
+	podName := os.Getenv("HOSTNAME")
+	if podName == "" {
+		podName = "unknown-pod-name"
+	}
+
+	// Get the actual pod namespace
+	podNamespaceBytes, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		podNamespaceBytes = []byte("unknown-pod-namespace")
+	}
+	podNamespace := string(podNamespaceBytes)
+
+	labels.podName = podName
+	labels.podNamespace = podNamespace
+	labels.mapId = fmt.Sprintf("%d", data.mapId)
+	labels.mapName = data.mapName
+	labels.mapType = data.mapType
+	return labels
+}
+
+func mapStruct2map(labels interface{}) map[string]string {
```

Comment:
Why not just use an existing function?
https://github.com/kmesh-net/kmesh/blob/22d9fdf4e46ff9a11ccc0eaf20b4d4c4c2ae34e8/pkg/controller/telemetry/metric.go#L583-L603